back/app/services/permissions/idea_permissions_service.rb

```diff
@@ -1,40 +1,41 @@
-# frozen_string_literal: true
-
 module Permissions
-  class IdeaPermissionsService < ProjectPermissionsService
+  class IdeaPermissionsService < BasePermissionsService
```

Comment:
I opted for delegation instead of inheritance, resulting in less "coupling", because the signatures are not entirely consistent, in the sense that the types of the parameters are different (idea vs. project).

In the case of the phase permissions, it even seems necessary, because different services are instantiated with different phases.

Reply:
Looking through the code, I can't help but feel that this has made everything more complicated again, I'd think I'd like to discuss your reasoning here.